### DISCUSSION TITLE
Error while installing aiida-core

### DISCUSSION THREAD

**Discussion by Ezio-droid** [aiidateam/aiida-core#5345] [Q&A]
Hello everyone,

I am trying to install aiida-core on my mackbook (macos- Monterey) version 12.2. But I am getting the following error:-

  error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  WARNING: No metadata found in /Users/tanaysahu/envs/aiida/lib/python3.9/site-packages
  Rolling back uninstall of psycopg2-binary
  Moving to /Users/tanaysahu/envs/aiida/lib/python3.9/site-packages/psycopg2/
   from /Users/tanaysahu/envs/aiida/lib/python3.9/site-packages/~sycopg2
  Moving to /Users/tanaysahu/envs/aiida/lib/python3.9/site-packages/psycopg2_binary-2.9.3-py3.9.egg-info
   from /Users/tanaysahu/envs/aiida/lib/python3.9/site-packages/~sycopg2_binary-2.9.3-py3.9.egg-info
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> psycopg2-binary

I tried different methods of installation as mentioned in the documentation like using pip, conda and core. But the error is same in all the cases. Any help in this regard would be appreciated.

Thanks

**Comment by zhubonan** [comment]
seems to be related to `psycopg2_binary` - you can try install 2.8.3 with pip first (e.g. `pip install psycopg2-binary==2.8.3`) then install aiida-core 1.6.5.

**Reply by ltalirz** [reply]
Your error message doesn't include the compiler error, which would be important to know why the wheel could not be built.

If Bonan's suggestion didn't fix things for you, you [may want to try](https://stackoverflow.com/a/39800677/1069467) `xcode-select --install`

**Reply by Ezio-droid** [reply]
I already tried the Bonan's suggestion before posting the issue here. When I try to install psycopg2_binary before and then aiida, aiida would automatically uninstall the previous package and do reinstallation and results in the same error.

But anyway I have shifted from Mac to redhat and it installed there sucessfully.
